client/src/main/java/org/threadly/litesockets/client/http/HTTPClient.java

```diff
@@ -0,0 +1,406 @@
+  }
+  
+  public HTTPResponseData request(final HTTPAddress ha, final HTTPRequest request, final ByteBuffer body, final TimeUnit tu, final long time) 
+      throws HTTPParsingException {
+    HTTPResponseData hr = null;
+    try {
+      hr = requestAsync(ha, request, body, tu, time).get();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+    } catch (Exception e) {
+      if(e.getCause() instanceof HTTPParsingException) {
+        throw (HTTPParsingException)e.getCause();
+      } else {
+        throw new HTTPParsingException(e.getCause());
+      }
+    }
+    return hr;
+  }
+
+  public ListenableFuture<HTTPResponseData> requestAsync(URL url) {
+    boolean ssl = false;
+    int port = HTTPConstants.DEFAULT_HTTP_PORT;
+    String host = url.getHost();
+    if(url.getProtocol().equalsIgnoreCase("https")) {
+      port = HTTPConstants.DEFAULT_HTTPS_PORT;
+      ssl = true;
+    }
+    if(url.getPort() != -1) {
+      port = url.getPort();
+    }
+    return requestAsync(new HTTPAddress(host, port, ssl), new HTTPRequestBuilder(url).build());
+  }
+  
+  public ListenableFuture<HTTPResponseData> requestAsync(final HTTPAddress ha, final HTTPRequest request) {
+    return requestAsync(ha, request, EMPTY_BUFFER);
+  }
+  
+  public ListenableFuture<HTTPResponseData> requestAsync(final HTTPAddress ha, final HTTPRequest request, final ByteBuffer body) {
+    return requestAsync(ha, request, body, TimeUnit.MILLISECONDS, defaultTimeout);
+  }
+  
+  public ListenableFuture<HTTPResponseData> requestAsync(final HTTPAddress ha, final HTTPRequest request, 
+      final ByteBuffer body, final TimeUnit tu, final long time) {
+    HTTPRequestWrapper hrw = new HTTPRequestWrapper(request, ha, body, tu.toMillis(time));
+    final ListenableFuture<HTTPResponseData> lf = hrw.slf;
+    queue.add(hrw);
+    if(ntse != null) {
+      ntse.wakeup();
+      if(isRunning.compareAndSet(false, true)) {
+        ssi.execute(new RunSocket());
+      }
+    } else {
+      processQueue();
+    }
+    return lf;
+  }
+  
+  private void processQueue() {
+    //This should be done after we do a .select on the ntse to check for more jobs before it exits.
+    while(maxConcurrent > inProcess.size()  && !queue.isEmpty()) {
+      HTTPRequestWrapper hrw = queue.poll();
+      //This runs concurrently if using a threaded SocketExecuter, so we have to null check before we add.
+      if(hrw != null) {
+        try {
+          hrw.client = getTCPClient(hrw.ha);
+          inProcess.put(hrw.client, hrw);
+          startWrite(hrw);
+        } catch (IOException e) {
+          hrw.slf.setFailure(e);
+        }
+      }
+    }
+  }
+
+  /**
+   * Stops the HttpClient.  Any requests currently pending will not be ran.  If this is not called
+   * and HTTPClient looses all references it will be GCed correctly. 
+   */
+  public void stop() {
+    if(ntse != null) {
+      ntse.stopIfRunning();
+      ntse.wakeup();
```

Comment:
yeah, I think this is left over from bad behavor in the stop on the ntse, should not need this anymore